src/algebra/fields/fp2.tcc

```diff
@@ -145,6 +145,10 @@ Fp2_model<n,modulus> Fp2_model<n,modulus>::Frobenius_map(unsigned long power) co
 template<mp_size_t n, const bigint<n>& modulus>
 Fp2_model<n,modulus> Fp2_model<n,modulus>::sqrt() const
 {
+    if (is_zero()) {
+        return *this;
+    }
+
```

Comment:
Fp3 had the same problem (for consistency, we should also remove the ``#if DEBUG`` around the Euler check on that one). I don't think we ever call sqrt on an Fp3 instance, but it's ugly for it to be inconsistent.